### PR TITLE
arch/x86: remove support for CONFIG_REALMODE

### LINKED ISSUE #17166: arch/x86: eliminate support for CONFIG_REALMODE
Jailhouse was the only "board" that started in real mode, so this option and its supporting logic in crt0.S and the linker script is orphaned and should be removed.

### PR DESCRIPTION
**Pull request by ghost**
We no longer support any platforms that bootstrap from real mode.

Fixes: #17166

Signed-off-by: Charles E. Youse <charles.youse@intel.com>

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]
LGTM

IIRC we only had this for jailhouse support (now removed) and early ACRN enabling.

File: arch/x86/core/ia32/crt0.S [hunk 1]
```diff
@@ -44,100 +44,6 @@
...
-	.pushsection ".rodata"
-
-	.align(4096)
-
-	.popsection
-#endif /* CONFIG_REALMODE */
-
-	/* processor is executing in 32-bit protected mode */
-
-	.balign 16,0x90
```
Comment (ioannisg) [inline arch/x86/core/ia32/crt0.S]: @gnuless this line is outside the CONFIG_REALMODE block

Comment (ioannisg) [inline arch/x86/core/ia32/crt0.S]: @gnuless this line is outside the CONFIG_REALMODE block

Comment (gnuless) [reply arch/x86/core/ia32/crt0.S]: Yeah, but the alignment is unnecessary.

Comment (ioannisg) [reply arch/x86/core/ia32/crt0.S]: OK, I just wanted to double-check with you

**Review by ioannisg — APPROVED** [review]

**Comment by dcpleung** [comment]
Just to make sure, booting from GRUB works too?

**Comment by gnuless** [comment]
> Just to make sure, booting from GRUB works too?

Yep.  We don't use `CONFIG_REALMODE` with GRUB, since it's nice enough to enter 32-bit protected mode before transferring control to our entry point.

**Review by dcpleung — APPROVED** [review]

**Review by wentongwu — APPROVED** [review]

**Pull request merged**